tests/phpunit/CiviTest/CiviUnitTestCase.php

```diff
@@ -65,6 +65,11 @@ class CiviUnitTestCase extends PHPUnit_Extensions_Database_TestCase {
   private static $dbInit = FALSE;
 
   /**
+   * Does the current setup support ONLY_FULL_GROUP_BY mode
+   */
+  protected $_supportFullGroupBy;
```

Comment:
@monishdeb @seamuslee001  OK, let me see if I follow correctly... this is "adding a protected variable that can be used to decide whether to run/skip ONLY_FULL_GROUP_BY specific test(s)." (I don't see any here now, but maybe this is looking ahead? I'll assume that premise valid.)

The simplest thing would be for those tests to just call `CRM_Utils_SQL::supportsFullGroupBy();` whenever they want, but clearly this does something a bit more complicated. So there must be a reason. Maybe it's a performance concern -- eg *if there's a lot of these tests, then we don't want to run `supportsFullGroupBy()` too often! We'll just do it once during setup.* But if that's the case, then this seems backwards to me. Compare the number of invocations between these:

 1. In the suite of 6000 unit-tests, a handful of specific tests call `supportsFullGroupBy()` and then drop-out.
 2. In the suite of 6000 unit-tests,  every one of them extends the `setUp()` procedure, which always calls `supportsFullGroupBy()`.

So my only theory/rationalization so far is bogus. Can you explain a little better why we need a new property in the base class?

Reply:
In my mind this doesn't add much extra overhead to the code base, it will allow any test to alter its behavior based on if it can or cannot the full group by mode.  Note it is used now in L390 of `tests/phpunit/CRM/Core/DAOTest.php`

Reply:
That's fine that it doesn't add a much overhead. But *why was done in the first place*? Why does this piece merit special attention in the `setUp()`?

Reply:
I'll have to leave that to @monishdeb as i followed his advice

Reply:
@totten it was my suggestion, to include this new property to base class in order to make it available for unit tests to run fullGroupBy specific queries. I asked myself "like ```$_dbconn``` what if we add a new property to tell whether the Mysql support fullGroupBy mode or not" but I agree with your point that it would cause an unnecessary burden for tests that extends the ```setUp()```. So I will revert this change in @seamuslee001 PR #11027 or should I create a separate PR for that ?

Reply:
Submitted a new PR https://github.com/civicrm/civicrm-core/pull/11208